src/js/binaryen.js-post.js

```diff
@@ -1098,13 +1108,25 @@
     this['optimize'] = function() {
       return Module['_BinaryenModuleOptimize'](module);
     };
+    this['optimizeFunction'] = function(func) {
+      if (typeof func === "string") func = this['getFunction'](func);
```

Comment:
this overloading could also work in `removeFunction`? Could be nice there too as a followup maybe.

Reply:
Not yet because, unlike `BinaryenFunctionOptimize`, `BinaryenRemoveFunction` expects a name, not a reference. What I'm trying to establish, though, is using names in `BinaryenAddSomething`-style functions, and `Something`-references in `BinaryenSomethingAdd`-style ones. Here, that'd be `BinaryenFunctionRemove` (note: doesn't fit well to the C++ API because `wasm->removeFunction` expects a name).

Reply:
Hmm, yeah. But we could eventually do the reverse, maybe: the C API expects a name, so if JS receives a name, pass that through. And if you get a number (a pointer), you can call `FunctionGetName` (which returns the name, parallel to `GetBody`), and use that.

Reply:
When doing the reverse, that is optionally resolving a reference to its name everywhere, we'll end up decoding and encoding even more strings :(

Reply:
Yeah, it wouldn't be optimal.